### PR TITLE
demo complete!

### PR DESCRIPTION
**Pull request by wnsgur4322**
please check demo

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
-  return (
-    <div>
-      <Navigation />
-      {/* <Home /> */
-      <SimpleFooter /> }
-    </div>
-  );
-}
\ No newline at end of file
+class App extends Component {
```
Comment (laurajjiang) [inline src/App.js]: keep this as function components, not class components  - keeping it constant across new React expectations

File: src/App.js [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
+
+  render(){
+    return(
+      <div>
+      <header><Navigation/></header>
+      <TfVis/>
+      <footer><SimpleFooter/></footer>
```
Comment (laurajjiang) [inline src/App.js]: remove header/footer tags, it should automatically load as header/footer without extra html tags

File: src/components/tf.js [hunk 3]
```diff
@@ -0,0 +1,438 @@
...
+ * A class that fetches the sprited MNIST dataset and returns shuffled batches.
+ *
+ * NOTE: This will get much easier. For now, we do data fetching and
+ * manipulation manually.
+ */
+
+window.tf = tf;
+window.tfvis = tfvis;
+var data = 0;
+var model = 0;
```
Comment (laurajjiang) [inline src/components/tf.js]: use let instead of var (let/const > var)

File: src/components/tf.js [hunk 4]
```diff
@@ -0,0 +1,438 @@
...
+
+export class MnistData {
+  constructor() {
```
Comment (laurajjiang) [inline src/components/tf.js:32]: look to switch this to a function export

File: src/components/tf.js [hunk 5]
```diff
@@ -0,0 +1,438 @@
...
+}
+
+function initModel() {
+  console.log("initModel ...");
+  window.model = getModel();
+}
+
+// doesn't work (fix later)
+function example_button() {
+  document.querySelector('#show-examples').addEventListener('click', async (e) => {
```
Comment (laurajjiang) [inline src/components/tf.js]: i don't know if DOM manipulation is in line with React expectations, try to read more about it? doesn't necessarily have to be fixed in this PR

File: src/components/tf.js [hunk 6]
```diff
@@ -0,0 +1,438 @@
...
+      drawArea.appendChild(canvas);
+      imageTensor.dispose();
+      }
+    
+    e.target.disabled = true;
+  });
+}
+
+function visor() {
+  document.querySelector('#show-visor').addEventListener('click', () => {
```
Comment (laurajjiang) [inline src/components/tf.js]: same thing w/ DOM manipulation

File: src/components/tf.js [hunk 7]
```diff
@@ -0,0 +1,438 @@
...
+
+function show_confusion(){
+  document.querySelector('#show-confusion').addEventListener('click', async (e) => {
+    console.log("calling confusion chart function ..")
+    showConfusion()
+  });
+}
+
+document.addEventListener('DOMContentLoaded', function() {
+  initModel();
```
Comment (laurajjiang) [inline src/components/tf.js]: switch to React hook of useEffect()

File: yarn-error.log [hunk 8]
```diff
@@ -0,0 +1,12802 @@
+Arguments:
```
Comment (laurajjiang) [inline yarn-error.log]: remove this file

**Review by laurajjiang — CHANGES_REQUESTED** [review]

**Review by thuyvyng — CHANGES_REQUESTED** [review]
I think the reason that we have the double click error is because of the onclick()s and then the .addeventlistener() within the function i think we only need one, laura and i tested it out and i think that the problem so if we could switch to only use onclick()

**Pull request closed without merging**